pkg/cmd/pulumi/convert.go

```diff
@@ -135,9 +135,14 @@ func runConvert(cwd string, language string, outDir string, generateOnly bool) r
 	case "yaml": // nolint: goconst
 		projectGenerator = yamlgen.GenerateProject
 	case "pulumi", "pcl":
-		// since we don't need Eject to get the full program,
-		// we can just convert the YAML directly to PCL
-		return runConvertPcl(cwd, outDir)
+		if cmdutil.IsTruthy(os.Getenv("PULUMI_DEV")) {
+			// since we don't need Eject to get the full program,
+			// we can just convert the YAML directly to PCL
+			return runConvertPcl(cwd, outDir)
+		}
+
+		return result.Errorf("cannot generate programs for %q language", language)
```

Comment:
I think this can just be `fallthrough`, but I'm never sure exactly when that's valid.

Reply:
First time I read about this, added and tested. Works like a charm ✅